CASSANDRA.md

```diff
@@ -0,0 +1,543 @@
+  }
+
+  abstract class WeatherStationQueries extends WeatherStationCF("weather_station") {
+
+    def getAllByCountry(country: Country): Future[List[WeatherStation]] =
+      select.where(_.country eqs country).fetch()
+  }
+
+  class DB(ks: KeySpaceDef) extends DatabaseImpl(ks) {
+
+    object stations extends WeatherStationQueries with connector.Connector
+  }
+
+  val db = new DB(ContactPoint.local.keySpace("db"))
+
+  val result = db.stations.getAllByCountry(Country("UK"))
+
+  result.onComplete(_ => db.shutdown())
+}
+```
+
+We need to define a new `Column` type and use it while defining the data model.
+
+**Quill (v0.3.2-SNAPSHOT)**
+```scala
+import io.getquill._
+import io.getquill.naming._
+import io.getquill.sources.MappedEncoding
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object Quill extends App {
+
+  val db = source(new CassandraAsyncSourceConfig[SnakeCase]("DB") with NoQueryProbing)
+
+  case class Country(code: String) extends AnyVal
+
+  object Country {
+
+    implicit val decode: MappedEncoding[String, Country] = mappedEncoding[String, Country](Country(_))
+    implicit val encode: MappedEncoding[Country, String] = mappedEncoding[Country, String](_.code)
+  }
+  case class WeatherStation(country: Country, city: String, stationId: String, entry: Int, value: Int)
+
+  object WeatherStation {
+
+    val getAllByCountry = quote {
+      (country: Country) =>
+        query[WeatherStation].filter(_.country == country)
+    }
+  }
+
+  val result = db.run(WeatherStation.getAllByCountry)(Country("UK"))
+
+  result.onComplete(_ => db.close())
+}
+```
+
+We only need to define implicit encodings from/to `String`.
+
+## Non-blocking IO ##
+
+This section would allow us to compare the different options the libraries offer to do non-blocking IO.
+
+**Java Driver (v3.0.0)**
+
+The Datastax driver allows you to execute queries [asynchronously](https://github.com/datastax/java-driver/tree/2.1/manual/async), returning `ListenableFuture`s.
+   
+**Phantom (v1.22.0)**
+
+Phantom is asynchronous by default and all operations returns `Future`s. It also provides an implementation of the [reactive-streams](https://github.com/reactive-streams/reactive-streams-jvm) specification on top of Akka. 
+
+**Quill (v0.3.2-SNAPSHOT)**
+
+Quill provides blocking, asynchronous and streaming sources for Cassandra. The asynchronous source returns `Future`s on all operations. The streaming source uses [Monix](https://github.com/monixio/monix) to return an `Observable`. Monix is a reactive library compatible with the [reactive-streams](https://github.com/reactive-streams/reactive-streams-jvm) protocol.
+
+## Other considerations ##
+
+So far we have only compared this libraries from technical point of view, but there are other things you might want to take into account like 3rd party dependencies. As both Phantom and Quill depend on the Datastax Java Driver, we are going to pay attention to which additional dependencies each of them add.
```

Comment:
So far we have only compared this libraries from technical point of view, but there are other things you
=>
There other aspects the user